src/components/SearchResult/SearchResult.tsx

```diff
@@ -160,6 +186,119 @@ export const SearchResult: FC<SearchResultProps> = ({ title, found, keywords })
           }
         </div>
       </div>
+      <Modal open={!!detailedDoc && (variant === 'snippets')} onClose={handleDetailedClose}>
+        <div className={styles.modal}>
+          <div className="h-[15%] text-right mb-1">
+            <button className="hover:text-primary" onClick={handleDetailedClose}>
+              <XMarkIcon className="w-8 h-8" />
+            </button>
+          </div>
+          {/*<div className="overflow-auto">*/}
+          <div className="h-[85%] scrollbar-thin scrollbar-thumb-gray-700 scrollbar-track-gray-500 overflow-auto">
+            {detailedDoc && <DetailedDoc doc={detailedDoc} />}
+          </div>
+        </div>
+      </Modal>
     </div>
   )
 };
+
+const validatePageSize = (pageSize: number) => (pageSize > 0) ? pageSize : defaultPageSize;
+
+const isPositiveVote = (doc: Doc) => (doc.upvote !== undefined) && (doc.upvote > 0);
+const hasNoVote = (doc: Doc) => (doc.upvote === undefined) || (doc.upvote === 0);
+
+const calcMaxPage = (variant: ViewVariant, pageSize: number, list: Doc[]) => {
+  const maxPage = (variant === 'snippets')
+    ? 0
+    : (Math.ceil(list.filter(hasNoVote).length / pageSize) - 1);
```

Comment:
I guess it would be more suitable to calculate max page number for tiles with

``` tsx
Math.floor(list.filter(hasNoVote).length / pageSize)
```

Reply:
done